src/libraries/Directory.Build.targets

```diff
@@ -201,25 +203,6 @@
     </When>
   </Choose>
 
-  <PropertyGroup>
-    <BuildAnalyzerReferences>$(BuildProjectReferences)</BuildAnalyzerReferences>
-    <BuildAnalyzerReferences Condition="'$(BuildingInsideVisualStudio)' == 'true'">false</BuildAnalyzerReferences>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <!-- Ensure AnalyzerReference items are restored and built
-         The target framework of Analyzers has no relationship to that of the refrencing project,
-         so we don't apply TargetFramework filters nor do we pass in TargetFramework.
-         When BuildProjectReferences=false we make sure to set BuildReference=false to make
-         sure not to try to call GetTargetPath in the outerbuild of the analyzer project. -->
-    <ProjectReference Include="@(AnalyzerReference)"
```

Comment:
Why isn't this needed anymore?

Reply:
When you originally added this code, source generators unnecessarily multi-targeted (by using `TargetFramework` instead of `TargetFramework`). Since then, source generators were updated to not multi-target and we do have general protection [that makes sure that they only target netstandard2.0](https://github.com/dotnet/runtime/blob/35e47797a36e29475d28af61f0fa0fb3bf067838/src/libraries/Directory.Build.targets#L236-L237). The TargetFramework.SDK was updated [to support single tfms](https://github.com/dotnet/arcade/commit/93656ab81d1b7f4507edc971465ef2e640694e64) and same as in the SDK, drops the TargetFramework metadata from ProjectReference items in case it isn't required (to not cause unnecessary evaluations).

Reply:
Ok, I couldn't remember exactly why we added this and I tried to trace it back in history and couldn't see it.  I trust your judgement.  Maybe a few things to try for regression purposes:
1. Open the S.T.J sln and build it.
2. `dotnet build --no-dependencies System.Text.Json.csproj`
3. `dotnet pack System.Text.Json.csproj`
4. `dotnet pack ---no-build System.Text.Json.csproj`